frontend/app/src/connection/DefaultStreamlitEndpoints.ts

```diff
@@ -120,11 +138,18 @@ export class DefaultStreamlitEndpoints implements StreamlitEndpoints {
     const form = new FormData()
     form.append(file.name, file)
 
-    const headers: Record<string, string> = {}
+    let headers: Record<string, string> = {}
     if (this.jwtHeader !== undefined) {
       headers[this.jwtHeader.jwtHeaderName] = this.jwtHeader.jwtHeaderValue
```

Comment:
nit: Is there any other usage of the JWT token besides file uploade? If not, wouldn't it be possible to just send it via the `fileUploadClientConfig.headers` instead of handling it with a dedicated host message / prop.

Reply:
Yeah, good point; sorry for not highlighting it in the description!

Definitely yes, we will remove the jwtHeader and all related machinery as soon as this PR is merged. 
the only reason why deletion is not part of this PR is that the SiS team wants to backport these changes to the old version of Streamlit, and just to not create potential difficulties for cherry-picking; we want to cherry-pick it without jwtHeader deletion code, only with additions.